src/pages/Search/SearchMoneyRequestReportPage.tsx

```diff
@@ -78,6 +95,28 @@ function SearchMoneyRequestReportPage({route}: SearchMoneyRequestPageProps) {
 
     const reportID = report?.reportID;
 
+    // Prevents creating duplicate transaction threads for legacy transactions
+    const hasCreatedLegacyThreadRef = useRef(false);
+
+    // Get transaction from search snapshot if not available in main collections
+    const {snapshotTransaction, snapshotViolations} = useMemo(() => {
+        if (!snapshot?.data || Object.keys(allReportTransactions).length > 0) {
+            return {snapshotTransaction: undefined, snapshotViolations: undefined};
+        }
+
+        const transactionKey = Object.keys(snapshot.data).find((key) => key.startsWith(ONYXKEYS.COLLECTION.TRANSACTION));
+        if (!transactionKey) {
+            return {snapshotTransaction: undefined, snapshotViolations: undefined};
+        }
+
+        const snapshotData = snapshot.data as Record<string, unknown>;
+        const transaction = snapshotData[transactionKey] as Transaction;
+        const violationKey = `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transaction.transactionID}`;
+        const violations = snapshotData[violationKey] as TransactionViolations | undefined;
+
+        return {snapshotTransaction: transaction, snapshotViolations: violations};
+    }, [snapshot?.data, allReportTransactions]);
```

Comment:
I hope this is not going to have too negative impact but it woul dbe great if we could check that out @adhorodyski @TMisiukiewicz